### PR TITLE
Rewrite code of MedlineImporter

### PR DESCRIPTION
**Pull request by aqurilla**
This fixes https://github.com/JabRef/jabref/issues/9536 by rewriting the `org.jabref.logic.importer.fileformat.MedlineImporter` class to use a StAX-Parser, and removes all JAXB dependencies from the class. The corresponding xjc task is also removed from `build.gradle`.

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks! Looks good! can you take a look at https://github.com/JabRef/jabref/issues/4273 as well?

File: src/main/java/org/jabref/logic/importer/fileformat/medline/ArticleID.java
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/medline/ArticleID.java:1]: I was thinking whether there is really a need for these classes, but it seems, that is indeed necessary.

I quickly checked https://nipafx.dev/java-var-anonymous-classes-tricks/ (hint by @Siedlerchr), but I did not see a win making the code better.

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you so much for working on this! 🎉

I have micro nitpicks 😇 - otherwise, looks good!

File: src/main/java/org/jabref/logic/importer/fileformat/MedlineImporter.java [hunk 1]
```diff
@@ -272,373 +390,718 @@ private void parseBookArticle(PubmedBookArticle currentArticle, List<BibEntry> b
...
-        if (book.getAuthorList() != null) {
-            List<AuthorList> authorLists = book.getAuthorList();
-            // authorLists size should be one
-            if (authorLists.size() == 1) {
-                for (AuthorList authorList : authorLists) {
-                    handleAuthors(fields, authorList);
+    private void parsePubmedData(XMLStreamReader reader, Map<Field, String> fields, String startElement)
+            throws XMLStreamException {
+        String publicationStatus = "";
+        List<ArticleID> articleIDList = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (aqurilla) [reply src/main/java/org/jabref/logic/importer/fileformat/MedlineImporter.java:396]: Thanks for you feedback! :) I'll make these changes

**Comment by aqurilla** [comment]
I think the `AstrophysicsDataSystemTest` failed due to some network issues? I did not make any changes related to those files

**Comment by koppor** [comment]
Yes, their servers return server error:

    java.io.IOException: org.jabref.logic.importer.FetcherServerException: Encountered HTTP Status Code 504

**Comment by aqurilla** [comment]
I found that Abstract and Title xml elements containing italics, bold tags etc. were getting cut short, so updated the code to handle this. 
Agreed that MathML support should be a separate PR since it looks like it will require handling a large number of tags!

**Review by koppor — APPROVED** [review]
Think, it's an improvement and I go forward with merge.

Regarding the italics:

```html
The apolipoprotein E (
                        <i>APOE</i>) gene on chromosome
```

should get

```latex
The apolipoprotein E (\textit{APOE}) gene on chromosome
```

Since it are very rare cases of HTML tags, it should be "easy". Maybe, this can be implemented when converting MathML?

**Pull request merged**